packages/flutter/lib/src/material/material_button.dart

```diff
@@ -378,6 +393,7 @@ class MaterialButton extends StatelessWidget {
   void debugFillProperties(DiagnosticPropertiesBuilder properties) {
     super.debugFillProperties(properties);
     properties.add(ObjectFlagProperty<VoidCallback>('onPressed', onPressed, ifNull: 'disabled'));
+    properties.add(ObjectFlagProperty<VoidCallback>('onLongPress', onLongPress, ifNull: 'disabled'));
```

Comment:
I think `ifNull: 'disabled'` will no longer be accurate here, since both onPressed and onLongPress would have to be null at the same time for the button to truly be disabled

Reply:
Agree. I don't know what I should do. Never used those debug flags before. Can you give me a hint or show a documentation?

Reply:
I'm not entirely familiar either, but based on some quick reading on Diagnosticables, I think you could omit the `ifNull` properties of both `onPressed` and `onLongPress` and use a [FlagProperty](https://master-api.flutter.dev/flutter/foundation/FlagProperty-class.html) separately:
```
FlagProperty(
  'enabled',
  value: onLongPress != null || onPressed != null,
  ifFalse: 'disabled',
),
```

There is more info on Diagnosticables in the [API docs](https://master-api.flutter.dev/flutter/foundation/DiagnosticableMixin-mixin.html) if you're interested in its importance.

cc/ @HansMuller do you happen to know the best approach for this?

Reply:
Actually, if you were to used `FlagProperty`, it might make the `onPressed` and `onLongPress` `ObjectFlagProperty`s pointless. So, I think the right approach is to add the `FlagProperty` and remove:
```
    properties.add(ObjectFlagProperty<VoidCallback>('onPressed', onPressed, ifNull: 'disabled'));
    properties.add(ObjectFlagProperty<VoidCallback>('onLongPress', onLongPress, ifNull: 'disabled'));
```